subsys/bluetooth/host/l2cap_internal.h

```diff
@@ -381,3 +381,16 @@ void bt_l2cap_register_ecred_cb(const struct bt_l2cap_ecred_cb *cb);
 
 /* Returns a server if it exists for given psm. */
 struct bt_l2cap_server *bt_l2cap_server_lookup_psm(uint16_t psm);
+
+/** Override tx MTU (ie. max SDU payload size) for the given channel.
```

Comment:
This brief is confusing: You are stating that the MTU is the `max SDU payload size`, but in fact the MTU is the `Maximum Transmission Unit` (as you probably is aware of), and the MPS is the `max SDU payload size`

So while I think what you are doing here makes sense, this brief is just confusing as you are basically mixing the definition of MPS and MTU

Reply:
Yes I think we got confused, there are many MTU definitions for different layers. @theob-pro FYI for your MTU sample.
Just to make sure we understand correctly this time, MTU then refers to the maximum L2CAP PDU size, right?

Reply:
I'm not so sure anymore, the spec is badly written IMO.
![image](https://user-images.githubusercontent.com/68428270/212058860-a7e6c7ea-c82e-452e-8872-3e80cc658e78.png)
what does "SDU segment" mean here, does it mean an L2CAP PDU that makes part of an SDU, or does it mean a complete SDU.

Reply:
> Just to make sure we understand correctly this time, MTU then refers to the maximum L2CAP PDU size, right?

I would say so, yes. 

I complete agree that the spec is confusing. 

I also need to correct that MPS is the maximum **PDU** payload size, and not the SDU payload size I wrote earlier. 

So the MPS is the maximum **PDU payload** size, and the MTU is the maximum **SDU size**.

In any case, stating that `tx MTU (ie. max SDU payload size)` is still incorrect, and should be `tx MTU (ie. max SDU size)` IMO

Reply:
Is it fine if I just remove the stuff in parenthesis?

```suggestion
/** Override tx MTU for the given channel.
```